url6da19b5a7a_krakrak_Adventure_Game_tgz-pJ8-adventure_MainJ8/src/items/CompanionsSword.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package items;
+
+import NPC.Eorlund;
+import NPC.Killable;
+import NPC.Npc;
+import commands.Game;
+import commands.GameState;
+import commands.History;
+import java.util.List;
+import java.util.Random;
+
+/**
+ *
+ * @author Irakiza Krak
+ */
+public class CompanionsSword extends AbstractWeapon{
+    
+    private GameState gameState;
+    private Random rdm = new Random();
+    private Double removeNbLives;
+    private int nbLivesRemain;
+    
+    public CompanionsSword(){
+        super("Companion's Sword", "Large silver two-handed sword", "co sw");
+    }
+
+    @Override
+    public GameState use(Game game) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 37-46 into a separate method.